src/js/services/srv-widget.test.js

```diff
@@ -401,4 +402,49 @@ describe('widgetSrv', () => {
 		$scope.$digest() // processes promise
 		expect(_selectedWidgetSrv.notifyAccessDenied).toHaveBeenCalled()
 	})
+
+	it('rejects with a message when a widget is already locked', () => {
+		mockSendPromiseOnce(false)
+
+		let promiseSpy = jest.fn()
+		let promiseCatch = jest.fn()
+		_service
+			.lockWidget(1)
+			.then(promiseSpy)
+			.catch(promiseCatch)
+		$scope.$digest() // processes promise
+
+		expect(Materia.Coms.Json.send).toHaveBeenCalledWith('widget_instance_lock', [1])
+		expect(promiseSpy).not.toHaveBeenCalled()
+		expect(promiseCatch).toHaveBeenCalledWith(
```

Comment:
Interesting approach I hadn't thought of.  I think it's ok as is, however one shouldn't have to test that promiseSpy wasn't called - if promiseCatch ends up getting called promiseSpy will always be skipped.

FWIW jest has another way to test catching with `expect().rejects` https://jestjs.io/docs/en/tutorial-async#rejects.  Though I realize this is using angular's promise lib - which may not be compatible